File: go/id_utils.go
package webexmessagehandler

import (
	"encoding/base64"
	"fmt"
	"strings"
)

// ToRestID converts a Mercury activity UUID to a Webex REST API ID.
//
// Mercury uses raw UUIDs; the REST API uses base64-encoded
// "ciscospark://us/{type}/{uuid}" URIs.
//
// resourceType should be "MESSAGE", "PEOPLE", or "ROOM".
func ToRestID(uuid string, resourceType string) string {
	uri := fmt.Sprintf("ciscospark://us/%s/%s", resourceType, uuid)
	return base64.StdEncoding.EncodeToString([]byte(uri))
}

// FromRestID converts a Webex REST API ID back to a raw UUID.
func FromRestID(restID string) (string, error) {
	decoded, err := base64.StdEncoding.DecodeString(restID)
	if err != nil {
		return "", fmt.Errorf("invalid base64 in REST ID: %w", err)
	}
	s := string(decoded)
	lastSlash := strings.LastIndex(s, "/")
	if lastSlash == -1 {
		return "", fmt.Errorf("invalid REST ID format: %s", restID)
	}
	return s[lastSlash+1:], nil
}

File: python/src/webex_message_handler/id_utils.py
"""Convert between Mercury activity UUIDs and Webex REST API IDs."""

import base64


def to_rest_id(uuid: str, resource_type: str) -> str:
    """Convert a Mercury activity UUID to a Webex REST API ID.

    Mercury uses raw UUIDs; the REST API uses base64-encoded
    ``ciscospark://us/{type}/{uuid}`` URIs.

    Args:
        uuid: Mercury UUID (e.g. activity.id).
        resource_type: Resource type — 'MESSAGE', 'PEOPLE', or 'ROOM'.

    Returns:
        REST API–compatible ID string.
    """
    uri = f"ciscospark://us/{resource_type}/{uuid}"
    return base64.b64encode(uri.encode()).decode()


def from_rest_id(rest_id: str) -> str:
    """Convert a Webex REST API ID back to a raw UUID.

    Args:
        rest_id: Base64-encoded REST API ID.

    Returns:
        The raw UUID portion.

    Raises:
        ValueError: If the format is invalid.
    """
    decoded = base64.b64decode(rest_id).decode()
    last_slash = decoded.rfind("/")
    if last_slash == -1:
        raise ValueError(f"Invalid REST ID format: {rest_id}")
    return decoded[last_slash + 1 :]
